### PR TITLE
CephFS - Scale Test Improvement

### PR DESCRIPTION
**Pull request by hkadam134**
Removed individual files and replaced it with a common IO file and included a scale_utils that has all functions related to scale tests.

Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-JQEGIE/

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_scale/cephfs_scale_utils.py [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+        """
+        Create subvolumes and mount them using the specified mount type.
+
+        :param fs_util_v1: FsUtilsV1 object
+        :param clients: List of client objects
+        :param subvolume_count: Number of subvolumes to create
+        :param default_fs: Default file system name
+        :param subvolume_name: Base name for subvolumes
+        :param subvolume_group_name: Name of the subvolume group
+        :param mount_type: Type of mount to use ("fuse" or "kernel")
```
Comment (sumabai) [inline tests/cephfs/cephfs_scale/cephfs_scale_utils.py:65]: can we have coverage for nfs too?

File: tests/cephfs/cephfs_scale/cephfs_scale_utils.py [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+
+        :param mount_paths: List of dictionaries containing client and mount path
+        :param io_type: Type of IO operation ("smallfile" or "largefile")
+        :param io_operation: Operation type for smallfile (e.g., create, read, etc.)
+        :param io_threads: Number of threads for smallfile
+        :param io_file_size: File size for smallfile
+        :param io_files: Number of files for smallfile
+        """
+
+        def run_smallfile_io(client, mount_path):
```
Comment (sumabai) [inline tests/cephfs/cephfs_scale/cephfs_scale_utils.py:388]: Can we use other existing IO tools and cmds for IO like - dd, crefi, tar and untar cmds ?

File: tests/cephfs/cephfs_scale/cephfs_scale_utils.py (hunk 1 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_scale/cephfs_scale_utils.py:65]: NFS QE Team does NFS coverage, so I skipped adding NFS.

File: tests/cephfs/cephfs_scale/cephfs_scale_utils.py (hunk 2 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_scale/cephfs_scale_utils.py:388]: Yes, I am including fio for now. other will be added later during further enhancements

**Review by neha-gangadhar — COMMENTED** [review]
Looks good. Please run once all comments are addressed.

**Comment by hkadam134** [comment]
Latest Logs after adding FIO data also. 

Randomly selected 3 TCs from all 3 IO types and ran the script : 
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-KE46KQ/

**Review by AmarnatReddy — COMMENTED** [review]
LGTM

File: tests/cephfs/cephfs_utilsV1.py [hunk 3]
```diff
@@ -3928,3 +3928,18 @@ def parse_caps(self, output):
                 cap_value = parts[1].strip().strip('"')
                 caps[cap_type] = cap_value
         return caps
+
+    def enable_distributed_pin_on_subvolumes(
```
Comment (sumabai) [inline tests/cephfs/cephfs_utilsV1.py:3932]: We may also need to validate if pinning happens after enablement, reference - validate_pin_stats in https://github.com/red-hat-storage/cephci/blob/master/tests/cephfs/snapshot_clone/cg_snap_utils.py

**Review by sumabai — APPROVED** [review]

File: tests/cephfs/cephfs_utilsV1.py (hunk 3 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_utilsV1.py:3932]: Sure. Will take this as enhancement

**Comment by neha-gangadhar** [comment]
@hkadam134 Please include the suite file to squid release as well. Good to merge.

**Comment by hkadam134** [comment]
> @hkadam134 Please include the suite file to squid release as well. Good to merge.

Done. Updated.

**Review by neha-gangadhar — APPROVED** [review]
Approved

**Pull request merged**